pkg/helper/envvarutils.go

```diff
@@ -157,3 +161,55 @@ func EnvVarReconciler(desired []v1.EnvVar, existing *[]v1.EnvVar, envVar string)
 	}
 	return update
 }
+
+// check if the secret ssl certs are populated and sets the path if they are
+// system-app and zync use this function
+func TlsCertPresent(pathSslEnvVar string, secretName string, databaseTLSEnabled bool) string {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		fmt.Printf("clientTLS error, get config : %v", err)
+		return ""
+	}
+	clientTLS, err := client.New(cfg, client.Options{})
+	if err != nil {
+		fmt.Printf("clientTLS error, client create : %v", err)
+		return ""
+	}
+	namespace, _ := GetOperatorNamespace()
+	var path string
+	var sslEnvVar string
+
+	// Determine the paths and corresponding secret keys
+	switch pathSslEnvVar {
+	case "DATABASE_SSL_CA":
+		path = "/tls/ca.crt"
+		sslEnvVar = "DB_SSL_CA"
+	case "DATABASE_SSL_CERT":
+		path = "/tls/tls.crt"
+		sslEnvVar = "DB_SSL_CERT"
+	case "DATABASE_SSL_KEY":
+		path = "/tls/tls.key"
+		sslEnvVar = "DB_SSL_KEY"
+	default:
+		return ""
+	}
+
+	// check the secret exists
+	secret := v1.Secret{}
+	nn := types.NamespacedName{
+		Name:      secretName,
+		Namespace: namespace,
+	}
+	err = clientTLS.Get(context.TODO(), nn, &secret)
+	if err != nil {
+		fmt.Printf("clientTLS error, get secret : %v", err)
+		return ""
+	}
+
+	// Check if SSL_KEY, SSL_CERT, and SSL_CA are empty
+	// checks the cert is populated in the secret, if so populates the path in the volume mount
+	if sslCert, ok := secret.Data[sslEnvVar]; ok && len(sslCert) > 0 && databaseTLSEnabled {
```

Comment:
so we are only setting paths if the value is non-empty - and when it is empty, we are returning "".
In case of having 2/3 created successfully (with values) - am I understanding it correctly that we are still going to just mount them, with 2/3 having the right values and 1/3 having empty string as value?
If so, the TLS validation will fail on the component level anyway so I'm trying to understand what is the benefit of doing this check and if it would be better to do it earlier (at ha level), keys exist + values are non-empty = add to volume for component validation of TLS, if keys exist + values are empty - do not even attempt to mount since we know this is going to be incorrect?

Reply:
No Mounts will have been created already with the cert values from the secret. 

This called when we are creating the env var , so believe it is the correct place to set them as they don't exist prior to this. System and Zync can only consume the path to the cert and not the cert itself.
The flow secret has the cert, those certs gets mounted , we then create env var that have the path to the mounted cert. These env var are what is used by system and zync to connect to the db with TLS.

As for the empty value check this will be caught In the HA reconciller so may be redundant to have the check here. But it causes no harm